src/components/rating/rating.tsx

```diff
@@ -227,6 +226,24 @@ export class Rating implements LabelableComponent, FormComponent, InteractiveCom
     this.calciteRatingChange.emit({ value });
   }
 
+  private onKeyboardPressed = (event: KeyboardEvent): void => {
+    if (!this.required && (event.key === "Enter" || event.key === " ")) {
```

Comment:
There is a util you could use here to check if a key is an activation one (Enter or Space): https://github.com/Esri/calcite-components/blob/master/src/utils/key.ts#L1.